E0060 needs to be updated to new format

From: src/test/compile-fail/E0060.rs

Error E0060 needs a span_label, updating it from:

```
error[E0060]: this function takes at least 1 parameter but 0 parameters were supplied
  --> src/test/compile-fail/E0060.rs:16:14
   |
16 |     unsafe { printf(); } //~ ERROR E0060
   |              ^^^^^^^^
   |
   = note: the following parameter type was expected: *const u8
```

To:

```
error[E0060]: this function takes at least 1 parameter but 0 parameters were supplied
  --> src/test/compile-fail/E0060.rs:16:14
   |
16 |     unsafe { printf(); } //~ ERROR E0060
   |              ^^^^^^^^ expected at least 1 parameter
   |
   = note: the following parameter type was expected: *const u8
```

Bonus: There may be a better way of displaying the information from the note.  Could it be part of the label?

--- a/src/librustc_typeck/check/mod.rs
+++ b/src/librustc_typeck/check/mod.rs
@@ -2384,6 +2384,12 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                     arg_count,
                     if arg_count == 1 {" was"} else {"s were"}),
                 error_code);
+
+            err.span_label(sp, &format!("expected {}{} parameter{}",
+                                        if variadic {"at least "} else {""},
+                                        expected_count,
+                                        if expected_count == 1 {""} else {"s"}));
+
             let input_types = fn_inputs.iter().map(|i| format!("{:?}", i)).collect::<Vec<String>>();
             if input_types.len() > 0 {
                 err.note(&format!("the following parameter type{} expected: {}",

--- a/src/test/compile-fail/E0060.rs
+++ b/src/test/compile-fail/E0060.rs
@@ -13,7 +13,8 @@ extern "C" {
 }
 
 fn main() {
-    unsafe { printf(); } //~ ERROR E0060
-                         //~| NOTE expected at least 1 parameter
-                         //~| NOTE the following parameter type was expected
+    unsafe { printf(); } 
+    //~^ ERROR E0060
+    //~| NOTE expected at least 1 parameter
+    //~| NOTE the following parameter type was expected
 }

--- a/src/test/compile-fail/E0060.rs
+++ b/src/test/compile-fail/E0060.rs
@@ -13,7 +13,7 @@ extern "C" {
 }
 
 fn main() {
-    unsafe { printf(); } 
+    unsafe { printf(); }
     //~^ ERROR E0060
     //~| NOTE expected at least 1 parameter
     //~| NOTE the following parameter type was expected

--- a/src/test/compile-fail/E0060.rs
+++ b/src/test/compile-fail/E0060.rs
@@ -14,4 +14,6 @@ extern "C" {
 
 fn main() {
     unsafe { printf(); } //~ ERROR E0060
+                         //~| NOTE expected at least 1 parameter
+                         //~| NOTE the following parameter type was expected
 }

--- a/src/test/compile-fail/E0061.rs
+++ b/src/test/compile-fail/E0061.rs
@@ -11,7 +11,8 @@
 fn f(a: u16, b: &str) {}
 
 fn main() {
-    f(0); //~ ERROR E0061 
-          //~| NOTE expected 2 parameters
-          //~| NOTE the following parameter types were expected
+    f(0);
+    //~^ ERROR E0061 
+    //~| NOTE expected 2 parameters
+    //~| NOTE the following parameter types were expected
 }

--- a/src/test/compile-fail/E0061.rs
+++ b/src/test/compile-fail/E0061.rs
@@ -12,7 +12,7 @@ fn f(a: u16, b: &str) {}
 
 fn main() {
     f(0);
-    //~^ ERROR E0061 
+    //~^ ERROR E0061
     //~| NOTE expected 2 parameters
     //~| NOTE the following parameter types were expected
 }

--- a/src/test/compile-fail/E0061.rs
+++ b/src/test/compile-fail/E0061.rs
@@ -11,5 +11,7 @@
 fn f(a: u16, b: &str) {}
 
 fn main() {
-    f(0); //~ ERROR E0061
+    f(0); //~ ERROR E0061 
+          //~| NOTE expected 2 parameters
+          //~| NOTE the following parameter types were expected
 }

--- a/src/test/compile-fail/issue-18819.rs
+++ b/src/test/compile-fail/issue-18819.rs
@@ -23,7 +23,8 @@ fn print_x(_: &Foo<Item=bool>, extra: &str) {
 }
 
 fn main() {
-    print_x(X);  //~error this function takes 2 parameters but 1 parameter was supplied
-    //~^ NOTE the following parameter types were expected: &Foo<Item=bool>, &str
+    print_x(X); 
+    //~^ ERROR this function takes 2 parameters but 1 parameter was supplied
+    //~| NOTE the following parameter types were expected: &Foo<Item=bool>, &str
     //~| NOTE expected 2 parameters
 }

--- a/src/test/compile-fail/issue-18819.rs
+++ b/src/test/compile-fail/issue-18819.rs
@@ -23,7 +23,7 @@ fn print_x(_: &Foo<Item=bool>, extra: &str) {
 }
 
 fn main() {
-    print_x(X); 
+    print_x(X);
     //~^ ERROR this function takes 2 parameters but 1 parameter was supplied
     //~| NOTE the following parameter types were expected: &Foo<Item=bool>, &str
     //~| NOTE expected 2 parameters

--- a/src/test/compile-fail/issue-18819.rs
+++ b/src/test/compile-fail/issue-18819.rs
@@ -25,4 +25,5 @@ fn print_x(_: &Foo<Item=bool>, extra: &str) {
 fn main() {
     print_x(X);  //~error this function takes 2 parameters but 1 parameter was supplied
     //~^ NOTE the following parameter types were expected: &Foo<Item=bool>, &str
+    //~| NOTE expected 2 parameters
 }

--- a/src/test/compile-fail/issue-3044.rs
+++ b/src/test/compile-fail/issue-3044.rs
@@ -15,6 +15,6 @@ fn main() {
     });
     //~^^ ERROR this function takes 2 parameters but 1 parameter was supplied
     //~^^^ NOTE the following parameter types were expected
-    //
+    //~| NOTE expected 2 parameters
     // the first error is, um, non-ideal.
 }

--- a/src/test/compile-fail/issue-3044.rs
+++ b/src/test/compile-fail/issue-3044.rs
@@ -14,7 +14,7 @@ fn main() {
     needlesArr.iter().fold(|x, y| {
     });
     //~^^ ERROR this function takes 2 parameters but 1 parameter was supplied
-    //~^^^ NOTE the following parameter types were expected
+    //~| NOTE the following parameter types were expected
     //~| NOTE expected 2 parameters
     // the first error is, um, non-ideal.
 }

--- a/src/test/compile-fail/issue-4935.rs
+++ b/src/test/compile-fail/issue-4935.rs
@@ -13,3 +13,4 @@
 fn foo(a: usize) {}
 fn main() { foo(5, 6) } //~ ERROR this function takes 1 parameter but 2 parameters were supplied
 //~^ NOTE the following parameter type was expected
+//~| NOTE expected 1 parameter

--- a/src/test/compile-fail/issue-4935.rs
+++ b/src/test/compile-fail/issue-4935.rs
@@ -11,6 +11,7 @@
 // Regression test for issue #4935
 
 fn foo(a: usize) {}
-fn main() { foo(5, 6) } //~ ERROR this function takes 1 parameter but 2 parameters were supplied
-//~^ NOTE the following parameter type was expected
+fn main() { foo(5, 6) }
+//~^ ERROR this function takes 1 parameter but 2 parameters were supplied
+//~| NOTE the following parameter type was expected
 //~| NOTE expected 1 parameter

--- a/src/test/compile-fail/method-call-err-msg.rs
+++ b/src/test/compile-fail/method-call-err-msg.rs
@@ -20,10 +20,13 @@ impl Foo {
 fn main() {
     let x = Foo;
     x.zero(0)   //~ ERROR this function takes 0 parameters but 1 parameter was supplied
+     //~^ NOTE expected 0 parameters
      .one()     //~ ERROR this function takes 1 parameter but 0 parameters were supplied
      //~^ NOTE the following parameter type was expected
+     //~| NOTE expected 1 parameter
      .two(0);   //~ ERROR this function takes 2 parameters but 1 parameter was supplied
      //~^ NOTE the following parameter types were expected
+     //~| NOTE expected 2 parameters
 
     let y = Foo;
     y.zero()

--- a/src/test/compile-fail/not-enough-arguments.rs
+++ b/src/test/compile-fail/not-enough-arguments.rs
@@ -20,4 +20,5 @@ fn main() {
   foo(1, 2, 3);
   //~^ ERROR this function takes 4 parameters but 3
   //~^^ NOTE the following parameter types were expected
+  //~| NOTE expected 4 parameters
 }

--- a/src/test/compile-fail/not-enough-arguments.rs
+++ b/src/test/compile-fail/not-enough-arguments.rs
@@ -19,6 +19,6 @@ fn foo(a: isize, b: isize, c: isize, d:isize) {
 fn main() {
   foo(1, 2, 3);
   //~^ ERROR this function takes 4 parameters but 3
-  //~^^ NOTE the following parameter types were expected
+  //~| NOTE the following parameter types were expected
   //~| NOTE expected 4 parameters
 }

--- a/src/test/compile-fail/overloaded-calls-bad.rs
+++ b/src/test/compile-fail/overloaded-calls-bad.rs
@@ -42,7 +42,9 @@ fn main() {
     let ans = s();
     //~^ ERROR this function takes 1 parameter but 0 parameters were supplied
     //~| NOTE the following parameter type was expected
+    //~| NOTE expected 1 parameter
     let ans = s("burma", "shave");
     //~^ ERROR this function takes 1 parameter but 2 parameters were supplied
     //~| NOTE the following parameter type was expected
+    //~| NOTE expected 1 parameter
 }

--- a/src/test/compile-fail/variadic-ffi-3.rs
+++ b/src/test/compile-fail/variadic-ffi-3.rs
@@ -18,8 +18,10 @@ fn main() {
     unsafe {
         foo(); //~ ERROR: this function takes at least 2 parameters but 0 parameters were supplied
         //~^ NOTE the following parameter types were expected
+        //~| NOTE expected at least 2 parameters
         foo(1); //~ ERROR: this function takes at least 2 parameters but 1 parameter was supplied
         //~^ NOTE the following parameter types were expected
+        //~| NOTE expected at least 2 parameters
 
         let x: unsafe extern "C" fn(f: isize, x: u8) = foo;
         //~^ ERROR: mismatched types

